src/DynamoPackages/PackageLoader.cs

```diff
@@ -337,7 +378,21 @@ private void ScanAllPackageDirectories(IPreferences preferences)
         {
             foreach (var packagesDirectory in packagesDirectories)
             {
-                ScanPackageDirectories(packagesDirectory, preferences);
+
+                if (preferences is IDisablePackageLoadingPreferences disablePrefs
+                    &&
+                    //if this directory is the standard library location
+                    //and loading from there is disabled, don't scan the directory.
+                    ((disablePrefs.DisableStandardLibrary && packagesDirectory == StandardLibraryDirectory)
+                    //or if custom package directories are disabled, and this is a custom package directory, don't scan.
+                    || (disablePrefs.DisableCustomPackageLocations && preferences.CustomPackageFolders.Contains(packagesDirectory))))
+                {
+                    continue;
```

Comment:
I wonder if it might be a good idea to log a message in the Dynamo console here saying that so and so package location has been skipped because of the preference setting?